packages/flutter_tools/lib/src/isolated/devfs_web.dart

```diff
@@ -1004,17 +1011,49 @@ Future<Directory> _loadDwdsDirectory(FileSystem fileSystem, Logger logger) async
 }
 
 String _stripBasePath(String path, String basePath) {
-  while (path.startsWith('/')) {
-    path = path.substring(1);
-  }
+  path = _stripLeadingSlashes(path);
   if (path.startsWith(basePath)) {
     path = path.substring(basePath.length);
   } else {
     // The given path isn't under base path, return null to indicate that.
     return null;
   }
+  return _stripLeadingSlashes(path);
+}
+
+String _stripLeadingSlashes(String path) {
   while (path.startsWith('/')) {
     path = path.substring(1);
   }
   return path;
 }
+
+String _stripTrailingSlashes(String path) {
+  while (path.endsWith('/')) {
+    path = path.substring(0, path.length - 1);
+  }
+  return path;
+}
+
+String _parseBasePathFromIndexHtml(File indexHtml) {
+  final String htmlContent = indexHtml.existsSync()
+      ? indexHtml.readAsStringSync()
+      : _kDefaultIndex;
+
+  final Document document = parse(htmlContent);
+  final Element baseElement = document.querySelector('base');
+  String baseHref = baseElement?.attributes == null ? null : baseElement.attributes['href'];
+
+  if (baseHref == null) {
+    baseHref = '';
+    print('WARNING: Please add a <base> element to your index.html file.');
+  } else if (!baseHref.startsWith('/')) {
+    throw ToolExit('Base href in index.html must be an absolute path to work correctly. Found `${baseElement.outerHtml}`');
```

Comment:
For these, it would to give a longer message with some examples of good baseHrefs. 

```
Error: The base href in "web/index.html" must end with a "/", but found Found: `{outerhtml}`
. For example, to serve from the root use:

  <base href="/">

To serve from a subdomain (?) named "foo" use:

  <base href="/foo">


```

Plus some word smithing